client/components/gridicon/index.tsx

```diff
@@ -11,11 +10,19 @@ import {
 } from 'gridicons/dist/util/icons-offset';
 import spritePath from 'gridicons/svg-sprite/gridicons.svg';
 
-function needsOffset( name, icons ) {
+function needsOffset( name: string, icons: [string] ) {
 	return icons.indexOf( name ) >= 0;
 }
 
-const Gridicon = React.forwardRef( ( props, ref ) => {
+interface Props {
+	icon: string;
+	size?: number;
+	onClick?: ( event?: React.MouseEvent< SVGSVGElement, MouseEvent > ) => any;
```

Comment:
Slightly more concise: 

```suggestion
	onClick?: React.MouseEventHandler< SVGSVGElement >;
```

Reply:
Much more compact, thank you!